### PR TITLE
Add facade for CarSetup and Component

### PR DESCRIPTION
**Pull request by theodesjardins**

### CODE REVIEW

File: formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImpl.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public void deleteCarSetup(CarSetupDTO carSetupDTO) {
+        if (carSetupDTO == null) throw new IllegalArgumentException("null carSetupDTO, cannot delete");
+
+        carSetupService.delete(beanMappingService.mapTo(carSetupDTO, CarSetup.class));
+    }
+
+    @Override
+    public void addCarSetup(CarSetupDTO carSetupDTO) {
+        if (carSetupDTO == null) throw new IllegalArgumentException("null carSetupAddDTO, cannot add");
```
Comment (elderanakain) [inline formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImpl.java]: Change `carSetupAddDTO` to `carSetupDTO`.

File: formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImpl.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        CarSetup carSetup = beanMappingService.mapTo(carSetupDTO, CarSetup.class);
+        carSetupService.add(carSetup);
+        carSetupDTO.setId(carSetup.getId());
```
Comment (elderanakain) [inline formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImpl.java]: Why do you set an id here?

File: formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImpl.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        componentService.delete(beanMappingService.mapTo(componentDTO, Component.class));
+    }
+
+    @Override
+    public void addComponent(ComponentDTO componentDTO) {
+        if (componentDTO == null) throw new IllegalArgumentException("null componentAddDTO, cannot add");
+
+        Component component = beanMappingService.mapTo(componentDTO, Component.class);
+        componentService.add(component);
+        componentDTO.setId(component.getId());
```
Comment (elderanakain) [inline formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImpl.java]: Why do you set an id here?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        when(beanMappingServiceMock.mapTo(listComponent, ComponentDTO.class)).thenReturn(listDTOComponent);
+        //When
+        List<ComponentDTO> resListComponentDTO = new ArrayList<>(componentFacade.getAllComponent());
+        //Then
+        verify(componentService).getAllComponent();
+        assertEquals(resListComponentDTO.size(),1);
+        Assert.assertTrue(resListComponentDTO.contains(componentDTO));
+
+    }
+
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Unnecessary empty line.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        List<ComponentDTO> listDTOComponent = new ArrayList<>();
+        listDTOComponent.add(componentDTO);
+        when(beanMappingServiceMock.mapTo(listComponent, ComponentDTO.class)).thenReturn(listDTOComponent);
+        //When
+        List<ComponentDTO> resListComponentDTO = new ArrayList<>(componentFacade.getAllComponent());
+        //Then
+        verify(componentService).getAllComponent();
+        assertEquals(resListComponentDTO.size(),1);
+        Assert.assertTrue(resListComponentDTO.contains(componentDTO));
+
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Unnecessary empty line.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    public void getAllComponentTest() {
+
+        //Given
+        List<Component> listComponent = new ArrayList<>();
+        listComponent.add(testingComponent);
+        when(componentService.getAllComponent()).thenReturn(listComponent);
+        List<ComponentDTO> listDTOComponent = new ArrayList<>();
+        listDTOComponent.add(componentDTO);
+        when(beanMappingServiceMock.mapTo(listComponent, ComponentDTO.class)).thenReturn(listDTOComponent);
+        //When
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        List<ComponentDTO> resListComponentDTO = new ArrayList<>(componentFacade.getAllComponent());
+        //Then
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        //when
+        componentFacade.updateComponent(componentDTO);
+        //then
+        verify(componentService, times(1)).update(testingComponent);
+    }
+
+    @Test
+    public void addComponentTest() {
+
+        when(componentService.findById(5L)).thenReturn(testingComponent);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+        //when
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+        componentFacade.deleteComponent(componentDTO);
+        //then
+        verify(componentService, times(1)).delete(testingComponent);
+    }
+
+    @Test
+    public void updateComponentTest() {
+
+        when(componentService.findById(5L)).thenReturn(testingComponent);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+        //When
+        ComponentDTO resComponentDTO = componentFacade.findByID(testingComponent.getId());
+        //Then
+        assertEquals(resComponentDTO,componentDTO);
+    }
+
+    @Test
+    public void deleteComponentTest() {
+
+        when(componentService.findById(5L)).thenReturn(testingComponent);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    @Test
+    public void findComponentById() {
+        //Given
+        when(componentService.findById(testingComponent.getId())).thenReturn(testingComponent);
+        when(beanMappingServiceMock.mapTo(testingComponent, ComponentDTO.class)).thenReturn(componentDTO);
+        //When
+        ComponentDTO resComponentDTO = componentFacade.findByID(testingComponent.getId());
+        //Then
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+        //When
+        when(beanMappingServiceMock.mapTo(componentDTO,Component.class)).thenReturn(testingComponent);
+    }
+
+    @Test
+    public void findComponentById() {
+        //Given
+        when(componentService.findById(testingComponent.getId())).thenReturn(testingComponent);
+        when(beanMappingServiceMock.mapTo(testingComponent, ComponentDTO.class)).thenReturn(componentDTO);
+        //When
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+    public void setUp() {
+
+        componentDTO = new ComponentDTO();
+
+        componentDTO.setType(ComponentType.ENGINE);
+        componentDTO.setName("testWithEngine");
+
+        testingComponent = createComponent();
+
+        //When
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: You don't need this comment here.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+        List<CarSetupDTO> listDTOCarSetup = new ArrayList<>();
+        listDTOCarSetup.add(carSetupDTO);
+        when(beanMappingServiceMock.mapTo(listCarSetup, CarSetupDTO.class)).thenReturn(listDTOCarSetup);
+        //When
+        List<CarSetupDTO> resListCarSetupDTO = new ArrayList<>(carSetupFacade.getAllCarSetup());
+        //Then
+        verify(carSetupService).getAllCarSetup();
+        assertEquals(resListCarSetupDTO.size(),1);
+        Assert.assertTrue(resListCarSetupDTO.contains(carSetupDTO));
+
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Unnecessary empty line.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+        //Given
+        List<CarSetup> listCarSetup = new ArrayList<>();
+        listCarSetup.add(testingCarSetup);
+        when(carSetupService.getAllCarSetup()).thenReturn(listCarSetup);
+        List<CarSetupDTO> listDTOCarSetup = new ArrayList<>();
+        listDTOCarSetup.add(carSetupDTO);
+        when(beanMappingServiceMock.mapTo(listCarSetup, CarSetupDTO.class)).thenReturn(listDTOCarSetup);
+        //When
+        List<CarSetupDTO> resListCarSetupDTO = new ArrayList<>(carSetupFacade.getAllCarSetup());
+        //Then
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+    public void getAllCarSetupTest() {
+
+        //Given
+        List<CarSetup> listCarSetup = new ArrayList<>();
+        listCarSetup.add(testingCarSetup);
+        when(carSetupService.getAllCarSetup()).thenReturn(listCarSetup);
+        List<CarSetupDTO> listDTOCarSetup = new ArrayList<>();
+        listDTOCarSetup.add(carSetupDTO);
+        when(beanMappingServiceMock.mapTo(listCarSetup, CarSetupDTO.class)).thenReturn(listDTOCarSetup);
+        //When
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+        //when
+        carSetupFacade.updateCarSetup(carSetupDTO);
+        //then
+        verify(carSetupService, times(1)).update(testingCarSetup);
+    }
+
+    @Test
+    public void addCarSetupTest() {
+
+        when(carSetupService.findById(4L)).thenReturn(testingCarSetup);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+        //when
+        carSetupFacade.deleteCarSetup(carSetupDTO);
+        //then
+        verify(carSetupService, times(1)).delete(testingCarSetup);
+    }
+
+    @Test
+    public void updateCarSetupTest() {
+
+        when(carSetupService.findById(4L)).thenReturn(testingCarSetup);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+        //When
+        CarSetupDTO resCarSetupDTO = carSetupFacade.findByID(testingCarSetup.getId());
+        //Then
+        assertEquals(resCarSetupDTO,carSetupDTO);
+    }
+
+    @Test
+    public void deleteCarSetupTest() {
+
+        when(carSetupService.findById(4L)).thenReturn(testingCarSetup);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+        verify(carSetupService, times(1)).update(testingCarSetup);
+    }
+
+    @Test
+    public void addCarSetupTest() {
+
+        when(carSetupService.findById(4L)).thenReturn(testingCarSetup);
+        //when
+        carSetupFacade.addCarSetup(carSetupDTO);
+        //then
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+        carSetupFacade.updateCarSetup(carSetupDTO);
+        //then
+        verify(carSetupService, times(1)).update(testingCarSetup);
+    }
+
+    @Test
+    public void addCarSetupTest() {
+
+        when(carSetupService.findById(4L)).thenReturn(testingCarSetup);
+        //when
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+        verify(carSetupService, times(1)).delete(testingCarSetup);
+    }
+
+    @Test
+    public void updateCarSetupTest() {
+
+        when(carSetupService.findById(4L)).thenReturn(testingCarSetup);
+        //when
+        carSetupFacade.updateCarSetup(carSetupDTO);
+        //then
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+        assertEquals(resCarSetupDTO,carSetupDTO);
+    }
+
+    @Test
+    public void deleteCarSetupTest() {
+
+        when(carSetupService.findById(4L)).thenReturn(testingCarSetup);
+        //when
+        carSetupFacade.deleteCarSetup(carSetupDTO);
+        //then
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+        CarSetupDTO resCarSetupDTO = carSetupFacade.findByID(testingCarSetup.getId());
+        //Then
+        assertEquals(resCarSetupDTO,carSetupDTO);
+    }
+
+    @Test
+    public void deleteCarSetupTest() {
+
+        when(carSetupService.findById(4L)).thenReturn(testingCarSetup);
+        //when
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 26]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    @Test
+    public void findCarSetupById() {
+        //Given
+        when(carSetupService.findById(testingCarSetup.getId())).thenReturn(testingCarSetup);
+        when(beanMappingServiceMock.mapTo(testingCarSetup, CarSetupDTO.class)).thenReturn(carSetupDTO);
+        //When
+        CarSetupDTO resCarSetupDTO = carSetupFacade.findByID(testingCarSetup.getId());
+        //Then
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 27]
```diff
@@ -0,0 +1,116 @@
...
+        //When
+        when(beanMappingServiceMock.mapTo(carSetupDTO,CarSetup.class)).thenReturn(testingCarSetup);
+    }
+
+    @Test
+    public void findCarSetupById() {
+        //Given
+        when(carSetupService.findById(testingCarSetup.getId())).thenReturn(testingCarSetup);
+        when(beanMappingServiceMock.mapTo(testingCarSetup, CarSetupDTO.class)).thenReturn(carSetupDTO);
+        //When
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: Please add an empty line before this.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java [hunk 28]
```diff
@@ -0,0 +1,116 @@
...
+        carSetupDTO.setEngine(new ComponentDTO());
+        carSetupDTO.setSuspension(new ComponentDTO());
+        carSetupDTO.setBrakes(new ComponentDTO());
+        carSetupDTO.setCover(new ComponentDTO());
+        carSetupDTO.setTires(new ComponentDTO());
+        carSetupDTO.setTransmission(new ComponentDTO());
+
+        testingCarSetup = createCarSetup();
+
+        //When
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/CarSetupFacadeImplTest.java]: You don't need this comment here.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java (hunk 8 above)
Comment (theodesjardins) [reply formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/ComponentFacadeImplTest.java]: I don't know, I will delete it

**Pull request merged**